component/src/main/java/org/ballerinalang/net/kafka/KafkaUtils.java

```diff
@@ -0,0 +1,485 @@
+        addIntParamIfPresent(ProducerConfig.METRICS_SAMPLE_WINDOW_MS_CONFIG, bMap, configParams);
+        addIntParamIfPresent(ProducerConfig.METRICS_NUM_SAMPLES_CONFIG, bMap, configParams);
+        addStringParamIfPresent(ProducerConfig.METRICS_RECORDING_LEVEL_CONFIG, bMap, configParams);
+        addStringParamIfPresent(ProducerConfig.METRIC_REPORTER_CLASSES_CONFIG, bMap, configParams);
+        addIntParamIfPresent(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, bMap, configParams);
+        addIntParamIfPresent(ProducerConfig.CONNECTIONS_MAX_IDLE_MS_CONFIG, bMap, configParams);
+        addStringParamIfPresent(ProducerConfig.PARTITIONER_CLASS_CONFIG, bMap, configParams);
+        addStringParamIfPresent(ProducerConfig.INTERCEPTOR_CLASSES_CONFIG, bMap, configParams);
+        addIntParamIfPresent(ProducerConfig.TRANSACTION_TIMEOUT_CONFIG, bMap, configParams);
+
+        processDefaultProducerProperties(configParams);
+        return configParams;
+    }
+
+    public static void processDefaultConsumerProperties(Properties configParams) {
+        configParams.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, Constants.DEFAULT_KEY_DESERIALIZER);
+        configParams.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, Constants.DEFAULT_VALUE_DESERIALIZER);
+    }
+
+    public static void processDefaultProducerProperties(Properties configParams) {
+        configParams.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, Constants.DEFAULT_KEY_SERIALIZER);
+        configParams.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, Constants.DEFAULT_VALUE_SERIALIZER);
+    }
+
+    private static void processPropertiesArray(Annotation jmsConfig, Properties configParams) {
+        AnnAttrValue attributeValue = jmsConfig.getAnnAttrValue(Constants.PROPERTIES_ARRAY);
+        if (attributeValue != null) {
+            AnnAttrValue[] attributeValueArray = attributeValue.getAnnAttrValueArray();
+            for (AnnAttrValue annAttributeValue : attributeValueArray) {
+                String stringValue = annAttributeValue.getStringValue();
+                int index = stringValue.indexOf("=");
+                if (index != -1) {
+                    String key = stringValue.substring(0, index).trim();
+                    String value = stringValue.substring(index + 1).trim();
+                    configParams.put(key, value);
+                } else {
+                    throw new BallerinaException("Invalid " + Constants.PROPERTIES_ARRAY + " provided. Key value"
+                            + " pair is not separated by an '='");
+                }
+            }
+        }
+    }
+
+    private static void updateMappedParameters(Properties configParams) {
+        Iterator<Map.Entry<Object, Object>> iterator = configParams.entrySet().iterator();
+        Properties tempProps = new Properties();
+        while (iterator.hasNext()) {
+            Map.Entry<Object, Object> entry = iterator.next();
```

Comment:
Here iterator is used since we are removing some elements from map while iterating through it [1]
[1] https://dzone.com/articles/removing-entries-hashmap

Reply:
Yep, my bad, didn't see it.